fs/is-dir-with-files/src/index.ts

```diff
@@ -0,0 +1,35 @@
+import fs from 'node:fs/promises'
+
+const isFileOrDir = async (path: string) => {
+  try {
+    await fs.access(path)
+    return true
+  } catch (_) {
+    return false
+  }
+}
+
+const isDir = async (path: string) => {
+  const stats = await fs.stat(path)
+  return stats.isDirectory()
+}
+
+const isDirEmpty = async (path: string) => {
+  const files = await fs.readdir(path)
+  return files.length === 0
+}
+
+export const isDirWithFiles = async (path: string) => {
```

Comment:
The first name I thought of for this function is `isDirNotEmpty`, but that makes it seem like it necessarily takes in a path to a **directory**, rather than any path and return `false` if it's not a directory

I then thought about `isUnemptyDir` but that felt awkward, so I went with `isDirWithFiles`